clients/src/main/java/org/apache/kafka/common/utils/Java.java

```diff
@@ -38,7 +38,6 @@ static Version parseVersion(String versionString) {
 
     // Having these as static final provides the best opportunity for compilar optimization
     public static final boolean IS_JAVA9_COMPATIBLE = VERSION.isJava9Compatible();
-    public static final boolean IS_JAVA8_COMPATIBLE = VERSION.isJava8Compatible();
```

Comment:
@ijuma For me it seems `isJava8Compatible()` could also be removed (with all its test usages)

Reply:
Done.